### ISSUE TITLE
"implicitly hidden" error reported by ddc but not dart2js nor the analyzer

### ISSUE THREAD

**Issue by chalin** [dart-archive/dev_compiler#636]
Dart2js doesn't report any errors, but when I do `pub build --web-compiler=dartdevc` on the AngularDart doc example [toh-5](https://github.com/angular-examples/toh-5), I get:
```nocode
Building angular_tour_of_heroes...
Building dartdevc modules...
Error compiling dartdevc module:angular_tour_of_heroes|lib/lib__hero_detail_component.template.js
[error] Element 'Location' from SDK library 'dart:html' is implicitly hidden by 'location.dart'. (package:angular_tour_of_heroes/hero_detail_component.template.dart, line 251, col 84)
[error] Element 'Location' from SDK library 'dart:html' is implicitly hidden by 'location.dart'. (package:angular_tour_of_heroes/hero_detail_component.template.dart, line 252, col 54)

Please fix all errors before compiling (warnings are okay).
```
The above is an excerpt from [this Travis build log](https://travis-ci.org/dart-lang/site-webdev/jobs/239936772).

While I expect differences between dart2js and the ddc, this isn't a difference in behavior that I'd expect. Note that running the `dartanalyzer` over toh-5 results in `No issues found!`.

cc @kwalrath @kevmoo @vsmenon

**Comment by vsmenon** [comment]
This issue was moved to dart-lang/sdk#29803
